src/compiler/objective_c_plugin.cc

```diff
@@ -91,15 +125,13 @@ class ObjectiveCGrpcGenerator : public grpc::protobuf::compiler::CodeGenerator {
       return true;
     }
 
+    auto OmitIf = [](const ::grpc::string& s, const ::grpc::string& v) {
```

Comment:
The name `OmitIf` of this function is a little bit confusing. Something like `InvertedIf` sounds better to me.

In fact to make it cleaner, I think it might make sense to have `PreprocIf`, `PreprocIfNot`, `PreprocIfElse`, `PreprocIfNotElse`, remove the parameter `PreprocConditionFlag` from these functions, and always enable `kCheckIfDefined`.